torchrl/envs/gym_like.py

```diff
@@ -78,6 +78,7 @@ class GymLikeEnv(_EnvWrapper):
     @classmethod
     def __new__(cls, *args, **kwargs):
         cls._info_dict_reader = None
+        cls._batch_locked = True
         return super().__new__(cls, *args, **kwargs)
```

Comment:
same here I think it makes more sense to pass this as an arg rather than setting it and then checking if it's not set